--- a/3rdparty/assimp/assimp.pri
+++ b/3rdparty/assimp/assimp.pri
@@ -5,6 +5,8 @@
 CONFIG(debug, debug|release) : DEFINES+=_DEBUG
 CONFIG += exceptions
 
+DEFINES+=ASSIMP_BUILD_NO_OWN_ZLIB
+
 win32:DEFINES+=_CRT_SECURE_NO_WARNINGS
 
 VPATH += \
@@ -22,7 +24,6 @@ INCLUDEPATH += \
         $$PWD/include \
         $$PWD/include/Compiler \
         $$PWD/contrib/ConvertUTF \
-        $$PWD/contrib/zlib \
         $$PWD/contrib/irrXML \
         $$PWD/contrib/unzip
 
@@ -203,17 +204,6 @@ HEADERS += \
         contrib/unzip/crypt.h \
         contrib/unzip/ioapi.h \
         contrib/unzip/unzip.h \
-        contrib/zlib/crc32.h \
-        contrib/zlib/deflate.h \
-        contrib/zlib/inffast.h \
-        contrib/zlib/inffixed.h \
-        contrib/zlib/inflate.h \
-        contrib/zlib/inftrees.h \
-        contrib/zlib/trees.h \
-        contrib/zlib/zconf.h \
-        contrib/zlib/zconf.in.h \
-        contrib/zlib/zlib.h \
-        contrib/zlib/zutil.h \
         include/Compiler/poppack1.h \
         include/Compiler/pushpack1.h \
         code/BoostWorkaround/boost/foreach.hpp \
@@ -337,13 +327,4 @@ SOURCES += \
         contrib/ConvertUTF/ConvertUTF.c \
         contrib/irrXML/irrXML.cpp \
         contrib/unzip/ioapi.c \
-        contrib/unzip/unzip.c \
-        contrib/zlib/adler32.c \
-        contrib/zlib/compress.c \
-        contrib/zlib/crc32.c \
-        contrib/zlib/deflate.c \
-        contrib/zlib/inffast.c \
-        contrib/zlib/inflate.c \
-        contrib/zlib/inftrees.c \
-        contrib/zlib/trees.c \
-        contrib/zlib/zutil.c
+        contrib/unzip/unzip.c
--- a/3rdparty/assimp/code/makefile
+++ b/3rdparty/assimp/code/makefile
@@ -24,7 +24,7 @@ OBJECTS   += $(patsubst %.cpp,%.o,  $(wi
 OBJECTS   += $(patsubst %.cpp,%.o,  $(wildcard ./../contrib/irrXML/*.cpp))
 
 # C object files
-OBJECTSC  := $(patsubst %.c,%.oc,   $(wildcard ./../contrib/zlib/*.c))
+#OBJECTSC  := $(patsubst %.c,%.oc,   $(wildcard ./../contrib/zlib/*.c))
 OBJECTSC  += $(patsubst %.c,%.oc,   $(wildcard ./../contrib/ConvertUTF/*.c))
 OBJECTSC  += $(patsubst %.c,%.oc,   $(wildcard ./../contrib/unzip/*.c))
 
@@ -47,6 +47,8 @@ INCPATH = ../include
 # GCC compiler flags
 CPPFLAGS=-Wall
 
+DEFINEFLAGS+=ASSIMP_BUILD_NO_OWN_ZLIB
+
 # Setup environment for noboost build
 ifeq ($(NOBOOST),1)
     SINGLETHREADED = 1
@@ -85,7 +87,7 @@ STATIC = $(BINPATH)/libassimp$(NAMESUFFI
 all: $(SHARED_TARGET)
 
 $(SHARED_TARGET):  $(OBJECTS)  $(OBJECTSC)
-    gcc -o $@ $(OBJECTS) $(OBJECTSC) -shared -lstdc++
+    gcc -o $@ $(OBJECTS) $(OBJECTSC) -shared -lstdc++ -lz
 %.o:%.cpp
     $(CXX) -g -c  $(CPPFLAGS) $? -o $@ $(INCLUDEFLAGS) $(DEFINEFLAGS) -fPIC
 %.oc:%.c
--- a/qt3d.pro
+++ b/qt3d.pro
@@ -9,6 +9,8 @@ module_qt3d_tutorials.CONFIG = no_defaul
 
 !package: SUBDIRS += module_qt3d_tutorials
 
+LIBS += -lz
+
 gcov: SUBDIRS -= sub_tools
 
 # We need opengl, minimum es2
--- a/src/threed/threed.pro
+++ b/src/threed/threed.pro
@@ -7,6 +7,8 @@ load(qt_module)
 
 QMAKE_DOCS = $$PWD/doc/qt3d.qdocconf
 
+QMAKE_LFLAGS += -lz
+
 gcov {
     CONFIG += static
     QMAKE_CXXFLAGS += -fprofile-arcs -ftest-coverage
